agent_build_refactored/utils/aws/ec2.py

```diff
@@ -98,15 +96,45 @@ def _create_ssh_container(
             toolset_image_name
         )
 
+        # The AWS_PRIVATE_KEY must be saved inside the running container using an echo command with a pipe.
+        # Using a volume mount is not possible when docker is installed via snap and leaves a key file in the filesystem.
         cmd_args.extend(
-            additional_cmd_args
+            [     
+                "/bin/bash",
+                "-c",
+                f"mkdir /tmp/mounts; echo \"{os.environ['AWS_PRIVATE_KEY']}\" > {self._ssh_client_container_in_docker_private_key_path}; chmod 600 {self._ssh_client_container_in_docker_private_key_path}; cat {self._ssh_client_container_in_docker_private_key_path}; " +
+                command
+            ] 
         )
 
-        subprocess.run(
-            cmd_args,
-            check=True
+        logger.info(f"Running _{cmd_args}")
+        logger.info(
+            subprocess.check_output(
+                cmd_args
+            )
+        )
+
+        # Since the docker is run in the background, we need to wait for the bash command to be really excuted.
+        cmd = [
+                    "docker",
+                    "exec",
+                    "-i",
+                    container_name,
+                    "bash",
+                    "-c",
+                    f"while [ ! -f {self._ssh_client_container_in_docker_private_key_path} ]; do sleep 1; echo Waiting for the key file ...; done"
```

Comment:
Do we need some kind of timeout or similar? So we don't wait for ever in case key file is not written for whatever reason (I imagine that in case key file is not written / available in some short amount of time, this likely indicates developer error or similar and it doesn't make sense to try to wait any longer)